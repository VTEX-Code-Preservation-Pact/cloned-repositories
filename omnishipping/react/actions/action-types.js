export const CHANGE_ACTIVE_TAB = 'CHANGE_ACTIVE_TAB'
export const CHANGE_ACTIVE_SLA_OPTION = 'CHANGE_ACTIVE_SLA_OPTION'
export const SELECT_DELIVERY_WINDOW = 'SELECT_DELIVERY_WINDOW'

export const CHANGE_ACTIVE_PICKUP_OPTIONS = 'CHANGE_ACTIVE_PICKUP_OPTIONS'
export const CHANGE_ACTIVE_PICKUP_SELLER = 'CHANGE_ACTIVE_PICKUP_SELLER'
export const CHANGE_ACTIVE_DELIVERY_CHANNEL = 'CHANGE_ACTIVE_DELIVERY_CHANNEL'
export const CLOSE_PICKUP_MODAL = 'CLOSE_PICKUP_MODAL'
export const OPEN_PICKUP_MODAL = 'OPEN_PICKUP_MODAL'

export const CHANGE_ACTIVE_PICKUP_DETAILS = 'CHANGE_ACTIVE_PICKUP_DETAILS'
export const CLOSE_PICKUP_DETAILS = 'CLOSE_PICKUP_DETAILS'

export const UPDATE_ADDRESS_FORM = 'UPDATE_ADDRESS_FORM'
export const UPDATE_ADDRESS_RULES = 'UPDATE_ADDRESS_RULES'
export const CHANGE_SELECTED_ADDRESS_RULES = 'CHANGE_SELECTED_ADDRESS_RULES'
export const VALIDATE_ADDRESS_FORM = 'VALIDATE_ADDRESS_FORM'
export const VALIDATE_ADDRESS_FORM_FIELDS = 'VALIDATE_ADDRESS_FORM_FIELDS'
export const VALIDATE_POSTAL_CODE = 'VALIDATE_POSTAL_CODE'
export const ADD_ADDRESS_FORM_ADDRESS = 'ADD_ADDRESS_FORM_ADDRESS'
export const ADD_DELIVERY_ADDRESS = 'ADD_DELIVERY_ADDRESS'
export const UPDATE_ORDERFORM = 'UPDATE_ORDERFORM'

export const SELECT_ADDRESS_ITEM = 'SELECT_ADDRESS_ITEM'

export const EDIT_ADDRESS = 'EDIT_ADDRESS'
export const EDIT_OMNISHIPPING = 'EDIT_OMNISHIPPING'
export const NEW_ADDRESS = 'NEW_ADDRESS'
export const SHOW_ADDRESS_PRODUCTS = 'SHOW_ADDRESS_PRODUCTS'
export const SHOW_ADDRESS = 'SHOW_ADDRESS'
export const SHOW_SPLIT = 'SHOW_SPLIT'
export const SHOW_ADDRESS_LIST = 'SHOW_ADDRESS_LIST'
export const SHOW_PAYMENT_BUTTON = 'SHOW_PAYMENT_BUTTON'
export const SHOW_OMNISHIPPING_COMPLETED = 'SHOW_OMNISHIPPING_COMPLETED'
export const SHOW_OMNISHIPPING = 'SHOW_OMNISHIPPING'
export const ENABLE_OMNISHIPPING = 'ENABLE_OMNISHIPPING'
export const DISABLE_OMNISHIPPING = 'DISABLE_OMNISHIPPING'

export const UPDATE_SHIPPING_FAILURE = 'UPDATE_SHIPPING_FAILURE'
export const UPDATE_SHIPPING_FROM_SHIPPING_RATE_PREVIEW =
  'UPDATE_SHIPPING_FROM_SHIPPING_RATE_PREVIEW'
export const UPDATE_SHIPPING_REQUEST = 'UPDATE_SHIPPING_REQUEST'

export const SIMULATE_SHIPPING_REQUEST = 'SIMULATE_SHIPPING_REQUEST'
export const SIMULATE_SHIPPING_FAILURE = 'SIMULATE_SHIPPING_FAILURE'
export const SIMULATE_SHIPPING_SUCCESS = 'SIMULATE_SHIPPING_SUCCESS'

export const UPDATE_SIMULATION_OPTIONS = 'UPDATE_SIMULATION_OPTIONS'

export const REMOVE_ITEMS_REQUEST = 'REMOVE_ITEMS_REQUEST'

export const FOCUS_ON_INPUT = 'FOCUS_ON_INPUT'

export const HAS_PICKUP_OPTIONS_EVENT = 'HAS_PICKUP_OPTIONS_EVENT'
export const SELECTED_PICKUP_TAB_EVENT = 'SELECTED_PICKUP_TAB_EVENT'
export const SHIPPING_STATS_EVENT = 'SHIPPING_STATS_EVENT'
export const SEARCHED_PICKUP_ADDRESS_EVENT = 'SEARCHED_PICKUP_ADDRESS_EVENT'
export const SEARCHED_PICKUP_ADDRESS_GEOLOCATION_EVENT =
  'SEARCHED_PICKUP_ADDRESS_GEOLOCATION_EVENT'

export const VTEX_ID_STARTED = 'VTEX_ID_STARTED'
export const VTEX_ID_CLOSED = 'VTEX_ID_CLOSED'

export const CHANGE_ACTIVE_DELIVERY_PACKAGE = 'CHANGE_ACTIVE_DELIVERY_PACKAGE'

export const UPDATE_SELECTED_PACKAGE_SLIDER = 'UPDATE_SELECTED_PACKAGE_SLIDER'

export const HAS_GEOLOCATION = 'HAS_GEOLOCATION'

export const SELECT_SCHEDULED_DELIVERY = 'SELECT_SCHEDULED_DELIVERY'
export const TOGGLE_SCHEDULED_DELIVERY = 'TOGGLE_SCHEDULED_DELIVERY'

export const OMNISHIPPING_JS_ERROR = 'OMNISHIPPING_JS_ERROR'
export const OMNISHIPPING_API_ERROR = 'OMNISHIPPING_API_ERROR'

export const SET_ADITIONAL_SHIPPING_DATA = 'SET_ADITIONAL_SHIPPING_DATA'

export const OMNISHIPPING_PAGEVIEW_EVENT = 'OMNISHIPPING_PAGEVIEW_EVENT'

export const COMPLETE_OMNISHIPPING = 'COMPLETE_OMNISHIPPING'
