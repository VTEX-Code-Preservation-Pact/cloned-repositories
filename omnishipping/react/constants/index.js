export const PICKUP_IN_STORE = 'pickup-in-point'
export const DELIVERY = 'delivery'
export const PICKUP = 'pickup'
export const SCHEDULED_DELIVERY = 'Entrega Agendada'

export const RESIDENTIAL = 'residential'
export const COMMERCIAL = 'commercial'
export const SEARCH = 'search'
export const GIFT_REGISTRY = 'giftRegistry'

export const GEOCORDS = 'geoCoords'
export const ACCOUNT_INFO_ENDOPOINT =
  '/api/logistics/pub/shipping/configuration'
export const HIDE_MAP = 'HIDE_MAP'
export const SHOW_MAP = 'SHOW_MAP'
export const AVAILABLE = 'available'
export const SHIPPING_HASH = '#/shipping'
export const PAYMENT_HASH = '#/payment'

export const CHEAPEST = 'CHEAPEST'
export const FASTEST = 'FASTEST'
export const COMBINED = 'COMBINED'

export const DENIED = 'denied'

export const VTEXLOCAL = 'vtexlocal'
export const VTEXBETA = 'vtexcommercebeta'
export const VTEXSTABLE = 'vtexcommercestable'

export const DEFAULT_WIDTH = 50
export const DEFAULT_HEIGHT = 50

export const IMPORTANT = 'Important'
export const CRITICAL = 'Critical'
export const ERROR = 'Error'
export const INFO = 'Info'

export const OMNISHIPPING = 'omnishipping'
