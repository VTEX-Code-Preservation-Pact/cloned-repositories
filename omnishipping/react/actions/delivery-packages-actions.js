import * as types from './action-types'

export const updateSelectedPackageSlider = packageIndex => ({
  type: types.UPDATE_SELECTED_PACKAGE_SLIDER,
  packageIndex,
})
