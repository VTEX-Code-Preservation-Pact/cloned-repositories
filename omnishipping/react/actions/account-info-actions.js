import * as types from './action-types'
import {
  getAccountName,
  getLocale,
  getUserEmail,
} from '../selectors/order-form-selectors'
import { AUTHENTICATED_USER } from '../constants/checkout-events'
const $ = window.$

export const vtexIdStarted = () => ({
  type: types.VTEX_ID_STARTED,
})

export const closedVtexId = () => ({
  type: types.VTEX_ID_CLOSED,
})

export const startVtexID = callback => (dispatch, getState) => {
  const state = getState()

  dispatch(vtexIdStarted())

  window.vtexid &&
    window.vtexid.start({
      returnUrl: window.location.href,
      accountName: getAccountName(state) || window.vtex.accountName,
      locale: getLocale(state),
      email: getUserEmail(state),
    })

  const authCallback = () => callback && callback(true)
  if ($) {
    $(window).one(AUTHENTICATED_USER, authCallback)
    $(window).one('closed.vtexid', () => {
      dispatch(closedVtexId())
      $(window).off(AUTHENTICATED_USER, authCallback)
    })
  }
}
