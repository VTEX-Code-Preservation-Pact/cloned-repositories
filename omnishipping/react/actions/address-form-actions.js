import * as types from './action-types'
import { newAddress } from '../utils/AddressFormUtils'
import {
  getDefaultShipsTo,
  getStoreCountryCode,
} from '../selectors/order-form-selectors'
import { getSearchAddress } from '../selectors/address-form-selectors'
import { newAddressEvent } from './component-activity-actions'
import { SEARCH } from '../constants'
import { helpers } from 'vtex.address-form'
const { removeValidation } = helpers

export const updateAddressForm = ({
  address,
  searchAddress,
  channel,
  rules,
}) => ({
  type: types.UPDATE_ADDRESS_FORM,
  address,
  searchAddress,
  channel,
  rules,
})

export const validateAddressForm = rules => ({
  type: types.VALIDATE_ADDRESS_FORM,
  rules,
})

export const validateAddressFormFields = rules => ({
  type: types.VALIDATE_ADDRESS_FORM_FIELDS,
  rules,
})

export const validatePostalCode = rules => ({
  type: types.VALIDATE_POSTAL_CODE,
  rules,
})

export const addAddressFormAddress = ({ address, searchAddress, channel }) => ({
  type: types.ADD_ADDRESS_FORM_ADDRESS,
  address,
  searchAddress,
  channel,
})

export const focusOnInput = input => ({
  type: types.FOCUS_ON_INPUT,
  input,
})

export const addDeliveryAddress = () => (dispatch, getState) => {
  dispatch(newAddressEvent(true))
  const state = getState()
  const shipsTo = getDefaultShipsTo(state)
  const storeCountryCode = getStoreCountryCode(state)
  const channel = state.componentActivity.activeDeliveryChannel
  const searchAddress = removeValidation(getSearchAddress(state))
  const country =
    shipsTo.indexOf(storeCountryCode) > -1 ? storeCountryCode : shipsTo[0]
  dispatch(
    addAddressFormAddress({
      address: newAddress({ country }),
      searchAddress,
      channel,
    })
  )
}

export const addNewAddresses = () => (dispatch, getState) => {
  const state = getState()
  const shipsTo = getDefaultShipsTo(state)
  const storeCountryCode = getStoreCountryCode(state)
  const channel = state.componentActivity.activeDeliveryChannel
  const country =
    shipsTo.indexOf(storeCountryCode) > -1 ? storeCountryCode : shipsTo[0]
  dispatch(
    addAddressFormAddress({
      address: newAddress({ country }),
      searchAddress: newAddress({
        country,
        addressType: SEARCH,
      }),
      channel,
    })
  )
}
