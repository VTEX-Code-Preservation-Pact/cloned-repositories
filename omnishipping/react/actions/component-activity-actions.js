import * as types from './action-types'
import { setItem } from '../utils/LocalStorageUtils'
import { OMNISHIPPING } from '../constants'
import { SHIPPING_INFO_UPDATED } from '../constants/checkout-events'
import {
  changeActiveTab,
  updateShippingData,
} from '../actions/order-form-actions'
import { isPickup } from '../utils/DeliveryChannelsUtils'
import {
  selectedPickupEvent,
  shippingStatsEvent,
} from '../actions/metric-actions'
import { validateAddressFormFields } from './address-form-actions'
import { shouldCompleteOmnishipping } from '../selectors/omnishipping-selectors'
import { getShippingDataWithSelectedDelivery } from '../selectors/delivery-selectors'
import { isValidAddress } from '../selectors/address-form-selectors'

const $ = window.$

export const selectDeliveryChannel = channel => (dispatch, getState) => {
  const state = getState()

  const activeDeliveryChannel = state.componentActivity.activeDeliveryChannel

  if (channel === activeDeliveryChannel) return

  dispatch(changeActiveDeliveryChannel(channel))
  dispatch(
    setAditionalShippingData({
      activeTab: channel,
    })
  )
  dispatch(
    changeActiveTab({
      channel: channel,
      loading: false,
      shouldUpdate: true,
    })
  )

  if (isPickup(channel)) {
    dispatch(selectedPickupEvent())
  }
}

export const enableOmnishipping = () => ({
  type: types.ENABLE_OMNISHIPPING,
})

export const disableOmnishipping = () => ({
  type: types.DISABLE_OMNISHIPPING,
})

export const showAddressList = show => ({
  type: types.SHOW_ADDRESS_LIST,
  show,
})

export const showAddress = show => ({
  type: types.SHOW_ADDRESS,
  show,
})

export const showSplit = show => ({
  type: types.SHOW_SPLIT,
  show,
})

export const showOmniShippingCompleted = show => ({
  type: types.SHOW_OMNISHIPPING_COMPLETED,
  show,
})

export const showOmniShipping = show => ({
  type: types.SHOW_OMNISHIPPING,
  show,
})

export const editOmnishipping = edit => ({
  type: types.EDIT_OMNISHIPPING,
  edit,
})

export const editAddress = show => ({
  type: types.EDIT_ADDRESS,
  show,
})

export const newAddressEvent = show => ({
  type: types.NEW_ADDRESS,
  show,
})

export const changeActiveDeliveryChannel = deliveryChannel => ({
  type: types.CHANGE_ACTIVE_DELIVERY_CHANNEL,
  deliveryChannel,
})

export const setAditionalShippingDataAction = () => ({
  type: types.SET_ADITIONAL_SHIPPING_DATA,
})

export const completeOmnishippingAction = () => ({
  type: types.COMPLETE_OMNISHIPPING,
})

export const completeOmnishipping = rules => (dispatch, getState) => {
  const state = getState()
  const shouldComplete = shouldCompleteOmnishipping(state)
  const shippingData = getShippingDataWithSelectedDelivery(state)

  dispatch(completeOmnishippingAction())
  if (!isValidAddress(state)) {
    dispatch(validateAddressFormFields(rules))
  }

  if (shouldComplete) {
    dispatch(showOmniShippingCompleted(true))
    dispatch(editOmnishipping(false))
    dispatch(updateShippingData(shippingData))
    dispatch(shippingStatsEvent())
  }
}

export const setAditionalShippingData = ({
  activeTab,
  selectedLeanShippingOption,
  isScheduledDeliveryActive,
}) => (dispatch, getState) => {
  const state = getState()

  const localActiveTab =
    activeTab || state.componentActivity.activeDeliveryChannel
  const localSelectedLeanShippingOption =
    selectedLeanShippingOption || state.delivery.activeDeliveryOption
  const localIsScheduledDeliveryActive =
    isScheduledDeliveryActive || state.delivery.isScheduledDeliveryActive

  const aditionalShippingData = {
    activeTab: localActiveTab,
    selectedLeanShippingOption: localSelectedLeanShippingOption,
    isScheduledDeliveryActive: localIsScheduledDeliveryActive,
    originComponent: OMNISHIPPING,
  }

  dispatch(setAditionalShippingDataAction(aditionalShippingData))

  setItem('aditionalShippingData', JSON.stringify(aditionalShippingData))

  $ && $(window).trigger(SHIPPING_INFO_UPDATED)
}
