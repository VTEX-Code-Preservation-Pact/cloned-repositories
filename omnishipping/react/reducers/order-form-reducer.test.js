import * as types from '../actions/action-types'

import { DELIVERY, PICKUP_IN_STORE, RESIDENTIAL, SEARCH } from '../constants'

import { addValidation } from '@vtex/address-form'
import reducer from './order-form-reducer'

describe('OrderForm reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle UPDATE_ORDERFORM with only PICKUP', () => {
    const orderForm = {
      canEditData: true,
      clientPreferencesData: {},
      giftRegistryData: {},
      clientProfileData: {},
      loggedIn: false,
      orderFormId: 'test',
      salesChannel: '2',
      sellers: [],
      shippingData: {
        selectedAddresses: [
          {
            addressType: RESIDENTIAL,
          },
        ],
        address: {
          addressId: '10',
        },
        logisticsInfo: [
          {
            slas: [
              {
                id: 'Test',
              },
            ],
            deliveryChannels: [
              {
                id: PICKUP_IN_STORE,
              },
            ],
          },
        ],
      },
      items: [{}],
      storePreferencesData: {
        currencySymbol: '',
        currencyFormatInfo: {
          currencyDecimalDigits: '',
          currencyDecimalSeparator: '',
          currencyGroupSeparator: '',
          startsWithCurrencySymbol: '',
        },
      },
    }

    expect(
      reducer([], {
        type: types.UPDATE_ORDERFORM,
        orderForm,
      })
    ).toMatchSnapshot()
  })

  it('should handle UPDATE_ORDERFORM when has different pickup selected', () => {
    const orderForm = {
      canEditData: true,
      clientPreferencesData: {},
      giftRegistryData: {},
      clientProfileData: {},
      loggedIn: false,
      orderFormId: 'test',
      salesChannel: '2',
      sellers: [],
      shippingData: {
        selectedAddresses: [
          {
            addressType: RESIDENTIAL,
          },
        ],
        address: {
          addressId: '10',
        },
        logisticsInfo: [
          {
            selectedSla: 'Test',
            selectedDeliveryChannel: PICKUP_IN_STORE,
            slas: [
              {
                deliveryChannel: PICKUP_IN_STORE,
                id: 'Test',
              },
            ],
            deliveryChannels: [
              {
                id: PICKUP_IN_STORE,
              },
            ],
          },
          {
            selectedSla: 'Test 2',
            selectedDeliveryChannel: PICKUP_IN_STORE,
            slas: [
              {
                deliveryChannel: PICKUP_IN_STORE,
                id: 'Test 2',
              },
            ],
            deliveryChannels: [
              {
                id: PICKUP_IN_STORE,
              },
            ],
          },
        ],
      },
      items: [{}],
      storePreferencesData: {
        currencySymbol: '',
        currencyFormatInfo: {
          currencyDecimalDigits: '',
          currencyDecimalSeparator: '',
          currencyGroupSeparator: '',
          startsWithCurrencySymbol: '',
        },
      },
    }
    const defaultState = {}
    const action = {
      type: types.UPDATE_ORDERFORM,
      orderForm,
    }
    const resultState = reducer(defaultState, action)

    expect(resultState.shippingData.logisticsInfo[1].selectedSla).toBe(null)
    expect(
      resultState.shippingData.logisticsInfo[1].selectedDeliveryChannel
    ).toBe(null)
  })

  it('should handle UPDATE_ORDERFORM with delivery', () => {
    const orderForm = {
      canEditData: true,
      clientPreferencesData: {},
      giftRegistryData: {},
      clientProfileData: {},
      loggedIn: false,
      orderFormId: 'test',
      salesChannel: '2',
      sellers: [],
      shippingData: {
        selectedAddresses: [
          {
            addressType: RESIDENTIAL,
          },
        ],
        address: {
          addressId: '10',
        },
        logisticsInfo: [
          {
            slas: [
              {
                id: 'Test',
              },
            ],
            deliveryChannels: [
              {
                id: PICKUP_IN_STORE,
              },
              {
                id: DELIVERY,
              },
            ],
          },
        ],
      },
      items: [{}],
      storePreferencesData: {
        currencySymbol: '',
        currencyFormatInfo: {
          currencyDecimalDigits: '',
          currencyDecimalSeparator: '',
          currencyGroupSeparator: '',
          startsWithCurrencySymbol: '',
        },
      },
    }

    expect(
      reducer([], {
        type: types.UPDATE_ORDERFORM,
        orderForm,
      })
    ).toMatchSnapshot()
  })

  it('should handle UPDATE_ORDERFORM with validation', () => {
    const orderForm = {
      canEditData: true,
      clientPreferencesData: {},
      giftRegistryData: {},
      clientProfileData: {},
      loggedIn: false,
      orderFormId: 'test',
      salesChannel: '2',
      sellers: [],
      shippingData: {
        selectedAddresses: [
          {
            addressType: RESIDENTIAL,
          },
        ],
        address: addValidation({
          addressId: '10',
        }),
        logisticsInfo: [
          {
            slas: [
              {
                id: 'Test',
              },
            ],
            deliveryChannels: [
              {
                id: PICKUP_IN_STORE,
              },
              {
                id: DELIVERY,
              },
            ],
          },
        ],
      },
      items: [{}],
      storePreferencesData: {
        currencySymbol: '',
        currencyFormatInfo: {
          currencyDecimalDigits: '',
          currencyDecimalSeparator: '',
          currencyGroupSeparator: '',
          startsWithCurrencySymbol: '',
        },
      },
    }

    expect(
      reducer([], {
        type: types.UPDATE_ORDERFORM,
        orderForm,
      })
    ).toMatchSnapshot()
  })

  it('should handle UPDATE_ORDERFORM with without slas', () => {
    const orderForm = {
      canEditData: true,
      clientPreferencesData: {},
      giftRegistryData: {},
      clientProfileData: {},
      loggedIn: false,
      orderFormId: 'test',
      salesChannel: '2',
      sellers: [],
      hippingData: {
        selectedAddresses: [
          {
            addressType: RESIDENTIAL,
          },
        ],
        address: {
          addressId: {
            value: '10',
          },
        },
        logisticsInfo: [
          {
            slas: [],
            deliveryChannels: [
              {
                id: PICKUP_IN_STORE,
              },
              {
                id: DELIVERY,
              },
            ],
          },
        ],
      },
      items: [{}],
      storePreferencesData: {
        currencySymbol: '',
        currencyFormatInfo: {
          currencyDecimalDigits: '',
          currencyDecimalSeparator: '',
          currencyGroupSeparator: '',
          startsWithCurrencySymbol: '',
        },
      },
    }

    expect(
      reducer([], {
        type: types.UPDATE_ORDERFORM,
        orderForm,
        address: {
          addressId: '10',
        },
        searchAddress: {
          addressId: '11',
        },
      })
    ).toMatchSnapshot()
  })

  it("should handle UPDATE_ADDRESS_FORM if it's the same addressId", () => {
    const address = {
      addressId: 'master',
      addressType: 'residential',
      city: 'Rio de Janeiro',
      complement: '100',
      country: 'BRA',
      geoCoordinates: [],
      neighborhood: 'Tijuca',
      number: '',
      postalCode: '333333333',
      receiverName: 'bananinha',
      reference: null,
      state: 'RJ',
      street: 'Rua Andrade Neves',
    }
    const selectedAddress = {
      storePreferencesData: {
        countryCode: 'BRA',
      },
      shippingData: {
        logisticsInfo: [
          {
            selectedDeliveryChannel: DELIVERY,
          },
        ],
        selectedAddresses: [{ ...address, addressType: SEARCH }],
      },
    }
    expect(
      reducer(selectedAddress, {
        type: types.UPDATE_ADDRESS_FORM,
        address: addValidation({
          ...address,
          city: 'Niteroi',
        }),
        searchAddress: addValidation({
          ...address,
          addressId: '11',
          city: 'Niteroi',
        }),
      })
    ).toMatchSnapshot()
  })

  it("should handle UPDATE_ADDRESS_FORM if it's the different addressId", () => {
    const address = {
      addressId: 'master',
      addressType: 'residential',
      city: 'Rio de Janeiro',
      complement: '100',
      country: 'BRA',
      geoCoordinates: [],
      neighborhood: 'Tijuca',
      number: '',
      postalCode: '333333333',
      receiverName: 'bananinha',
      reference: null,
      state: 'RJ',
      street: 'Rua Andrade Neves',
    }
    const selectedAddress = {
      storePreferencesData: {
        countryCode: 'BRA',
      },
      canEditData: true,
      shippingData: {
        logisticsInfo: [
          {
            selectedDeliveryChannel: DELIVERY,
          },
        ],
        selectedAddresses: [address],
        availableAddresses: [address],
      },
    }
    expect(
      reducer(selectedAddress, {
        type: types.UPDATE_ADDRESS_FORM,
        address: addValidation({
          ...address,
          addressId: '123456',
        }),
        searchAddress: addValidation({
          ...address,
          addressId: '12341116',
        }),
      })
    ).toMatchSnapshot()
  })
})
