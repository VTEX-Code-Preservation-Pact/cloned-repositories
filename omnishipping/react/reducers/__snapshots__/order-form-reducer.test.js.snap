// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderForm reducer should handle UPDATE_ADDRESS_FORM if it's the different addressId 1`] = `
Object {
  "canEditData": true,
  "shippingData": Object {
    "availableAddresses": Array [
      Object {
        "addressId": "master",
        "addressType": "residential",
        "city": "Rio de Janeiro",
        "complement": "100",
        "country": "BRA",
        "geoCoordinates": Array [],
        "neighborhood": "Tijuca",
        "number": "",
        "postalCode": "333333333",
        "receiverName": "bananinha",
        "reference": null,
        "state": "RJ",
        "street": "Rua Andrade Neves",
      },
    ],
    "logisticsInfo": Array [
      Object {
        "addressId": "12341116",
        "selectedDeliveryChannel": "delivery",
      },
    ],
    "selectedAddresses": Array [
      Object {
        "addressId": "master",
        "addressType": "residential",
        "city": "Rio de Janeiro",
        "complement": "100",
        "country": "BRA",
        "geoCoordinates": Array [],
        "neighborhood": "Tijuca",
        "number": "",
        "postalCode": "333333333",
        "receiverName": "bananinha",
        "reference": null,
        "state": "RJ",
        "street": "Rua Andrade Neves",
      },
    ],
  },
  "storePreferencesData": Object {
    "countryCode": "BRA",
  },
}
`;

exports[`OrderForm reducer should handle UPDATE_ADDRESS_FORM if it's the same addressId 1`] = `
Object {
  "shippingData": Object {
    "logisticsInfo": Array [
      Object {
        "addressId": "11",
        "selectedDeliveryChannel": "delivery",
      },
    ],
    "selectedAddresses": Array [
      Object {
        "addressId": "master",
        "addressType": "search",
        "city": "Rio de Janeiro",
        "complement": "100",
        "country": "BRA",
        "geoCoordinates": Array [],
        "neighborhood": "Tijuca",
        "number": "",
        "postalCode": "333333333",
        "receiverName": "bananinha",
        "reference": null,
        "state": "RJ",
        "street": "Rua Andrade Neves",
      },
    ],
  },
  "storePreferencesData": Object {
    "countryCode": "BRA",
  },
}
`;

exports[`OrderForm reducer should handle UPDATE_ORDERFORM with delivery 1`] = `
Object {
  "canEditData": true,
  "clientPreferencesData": Object {},
  "clientProfileData": Object {},
  "giftRegistryData": Object {},
  "items": Array [
    Object {},
  ],
  "loggedIn": false,
  "marketingData": undefined,
  "orderFormId": "test",
  "salesChannel": "2",
  "selectableGifts": undefined,
  "sellers": Array [],
  "shippingData": Object {
    "address": Object {
      "addressId": "10",
    },
    "logisticsInfo": Array [
      Object {
        "addressId": undefined,
        "deliveryChannels": Array [
          Object {
            "id": "pickup-in-point",
          },
          Object {
            "id": "delivery",
          },
        ],
        "slas": Array [
          Object {
            "id": "Test",
          },
        ],
      },
    ],
    "selectedAddresses": Array [
      Object {
        "addressType": "residential",
      },
    ],
  },
  "storePreferencesData": Object {
    "currencyFormatInfo": Object {
      "currencyDecimalDigits": "",
      "currencyDecimalSeparator": "",
      "currencyGroupSeparator": "",
      "startsWithCurrencySymbol": "",
    },
    "currencySymbol": "",
  },
}
`;

exports[`OrderForm reducer should handle UPDATE_ORDERFORM with only PICKUP 1`] = `
Object {
  "canEditData": true,
  "clientPreferencesData": Object {},
  "clientProfileData": Object {},
  "giftRegistryData": Object {},
  "items": Array [
    Object {},
  ],
  "loggedIn": false,
  "marketingData": undefined,
  "orderFormId": "test",
  "salesChannel": "2",
  "selectableGifts": undefined,
  "sellers": Array [],
  "shippingData": Object {
    "address": Object {
      "addressId": "10",
    },
    "logisticsInfo": Array [
      Object {
        "addressId": undefined,
        "deliveryChannels": Array [
          Object {
            "id": "pickup-in-point",
          },
        ],
        "slas": Array [
          Object {
            "id": "Test",
          },
        ],
      },
    ],
    "selectedAddresses": Array [
      Object {
        "addressType": "residential",
      },
    ],
  },
  "storePreferencesData": Object {
    "currencyFormatInfo": Object {
      "currencyDecimalDigits": "",
      "currencyDecimalSeparator": "",
      "currencyGroupSeparator": "",
      "startsWithCurrencySymbol": "",
    },
    "currencySymbol": "",
  },
}
`;

exports[`OrderForm reducer should handle UPDATE_ORDERFORM with validation 1`] = `
Object {
  "canEditData": true,
  "clientPreferencesData": Object {},
  "clientProfileData": Object {},
  "giftRegistryData": Object {},
  "items": Array [
    Object {},
  ],
  "loggedIn": false,
  "marketingData": undefined,
  "orderFormId": "test",
  "salesChannel": "2",
  "selectableGifts": undefined,
  "sellers": Array [],
  "shippingData": Object {
    "address": Object {
      "addressId": Object {
        "value": "10",
      },
    },
    "logisticsInfo": Array [
      Object {
        "addressId": undefined,
        "deliveryChannels": Array [
          Object {
            "id": "pickup-in-point",
          },
          Object {
            "id": "delivery",
          },
        ],
        "slas": Array [
          Object {
            "id": "Test",
          },
        ],
      },
    ],
    "selectedAddresses": Array [
      Object {
        "addressType": "residential",
      },
    ],
  },
  "storePreferencesData": Object {
    "currencyFormatInfo": Object {
      "currencyDecimalDigits": "",
      "currencyDecimalSeparator": "",
      "currencyGroupSeparator": "",
      "startsWithCurrencySymbol": "",
    },
    "currencySymbol": "",
  },
}
`;

exports[`OrderForm reducer should handle UPDATE_ORDERFORM with without slas 1`] = `Array []`;

exports[`OrderForm reducer should return the initial state 1`] = `Object {}`;
